pan-domain-node/src/panda.ts

```diff
@@ -0,0 +1,111 @@
+import * as iniparser from 'iniparser';
+import * as cookie from 'cookie';
+
+import { httpGet, base64ToPEM, parseCookie, verifySignature, parseUser } from './utils';
+import { ValidateUserFn, AuthenticationStatus, User, AuthenticationResult } from './api';
+
+interface PublicKeyHolder {
+    key: string,
+    lastUpdated: Date
+}
+
+function fetchPublicKey(region: string, bucket: String, keyFile: String): Promise<PublicKeyHolder> {
+    const path = `https://s3.${region}.amazonaws.com/${bucket}/${keyFile}`;
```

Comment:
Does this mean this file is publicly accessible? 😱

Reply:
Yep and that's fine because it's assymetric crypto, like how PGP works.

Anyone can verify the signature using the public key but only those who have the private key can generate the signature.